controllers/cluster_status.go

```diff
@@ -395,7 +395,7 @@ func (r *ClusterReconciler) getPoolerIntegrationsNeeded(ctx context.Context,
 	pgbouncerPoolerIntegrations, err := r.getPgbouncerIntegrationStatus(ctx, cluster, poolers)
 
 	for _, pooler := range poolers.Items {
-		if !slices.Contains(pgbouncerPoolerIntegrations.Secrets, pooler.Name) {
+		if !slices.Contains(pgbouncerPoolerIntegrations.Secrets, pooler.GetAuthQuerySecretName()) {
```

Comment:
You are right about the missing -pooler, however this would defeat the purpose of the check, as it would always return true by definition. What we want to check here is that the default secret name is in the list, which is probably a too weak condition to start with.

Reply:
Oh oO Understood, this effectively checks all possible combinations 🤔 Would it make sense to check for `pooler.Name + DefaultPgBouncerPoolerSecretSuffix`? 🤔

Reply:
That would still mean a log line is printed for every reconciliation loop if the pooler auth is not automatically handled. Moving the logging to a place where we are able to tell if we actually created the secret and do that only on creation would be the optimal solution.